src/searchModal.js

```diff
@@ -206,5 +272,12 @@ export default function searchModalFactory(config) {
         section.append(infoMessage);
     }
 
-    return instance.init(config);
+    // return initialized instance of searchModal
+    return instance.init(
+        _.defaults(config, {
+            renderTo: 'body',
+            query: '',
+            searchOnInit: true
+        })
+    );
```

Comment:
The default values are handled by the component factory: `component(api, defaults)`.